cmake/findDependencies.cmake

```diff
@@ -34,10 +34,14 @@ endif()
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 if (NOT USE_MATCHCOMPILER_OPT MATCHES "Off")
-    find_package(PythonInterp)
-    if (NOT ${PYTHONINTERP_FOUND})
-        message(WARNING "No python interpreter found. Therefore, the match compiler is switched off.")
-        set(USE_MATCHCOMPILER_OPT "Off")
+    find_package(PythonInterp 3 QUIET)
+    if (NOT PYTHONINTERP_FOUND)
+        set(PYTHONINTERP_FOUND "")
+        find_package(PythonInterp 2.7 QUIET)
```

Comment:
feel free to drop support for python 2.7 in the match compiler. you can do it in a follow up PR.

Reply:
Alright.

But I still think we should drop Python 2.7 as a whole and not just in parts. We still have holes in the CI matrix with that and I'd rather remove stuff from it than add more.

Reply:
I hope we can still find some CI-runner that has python 2.7. I also hope that it doesn't cost much to keep some user-scripts compatible with python 2.7 for a little while.